core/include/log.h

```diff
@@ -73,7 +73,15 @@ enum {
 /**
  * @brief Log message if level <= LOG_LEVEL
  */
+#ifdef __clang__    /* following pragmas required for clang 3.8.0 */
+#define LOG(level, ...) \
+    _Pragma("clang diagnostic push") \
+    _Pragma("clang diagnostic ignored \"-Wtautological-compare\"") \
+    if (level <= LOG_LEVEL) log_write(level, __VA_ARGS__) \
```

Comment:
There's an error in the original line, but maybe you could fix it while you're at it: `level` should be parenthesized at all instances, or an invocation like `LOG(quite_serious ? LOG_WARN : LOG_INFO, "Something's bad");` would fail.

Also I would surround the whole block in `do { } while (0)` even if it should not™ be needed.